internal/backends/collection.go

```diff
@@ -96,7 +97,14 @@ type QueryResult struct {
 // It also can be used to close the returned iterator and free underlying resources,
 // but doing so is not necessary - the handler will do that anyway.
 //
-// Passed sort document should be already validated. If sort document is invalid, function panics.
+// Filter may be ignored, or safely applied partially or entirely.
+// Extra documents will be filtered out by the handler.
+//
+// Sort should have one of the following forms: nil, {}, {"$natural": int64(1)} or {"$natural": int64(-1)}.
+// Other field names are not supported.
+// If non-empty, it should be applied.
```

Comment:
@noisersup PTAL

Reply:
Okay, I was a bit worried that we remove and implement the same things all over again, though the need of simplification of code on the backend level is understandable, gonna remove support for any other sort

Reply:
Yeah, before, we had support for unsafe sorting under the experimental flag, but that flag is gone, but the sorting remained. That's practically a bug on that layer